### ISSUE TITLE
When the write! macro's result is chained stability warnings show up, and the expansion site is not reported

### ISSUE THREAD

**Issue by renato-zannon** [rust-lang/rust#22013]
Example:

``` rust
fn main() {
    let mut buffer = Vec::new();
    writeln!(&mut buffer, "foo").unwrap();
}
```

This errors as:

```
<std macros>:2:20: 2:66 warning: use of unstable library feature 'io'
<std macros>:2 ( & mut * $ dst ) . write_fmt ( format_args ! ( $ ( $ arg ) * ) ) )
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<std macros>:2:20: 2:66 help: add #![feature(io)] to the crate attributes to silence this warning
<std macros>:2 ( & mut * $ dst ) . write_fmt ( format_args ! ( $ ( $ arg ) * ) ) )
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Contrast that with the error when the `.unwrap()` is removed:

```
<std macros>:2:1: 2:66 warning: unused result which must be used, #[warn(unused_must_use)] on by default
<std macros>:2 ( & mut * $ dst ) . write_fmt ( format_args ! ( $ ( $ arg ) * ) ) )
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<std macros>:1:1: 2:68 note: in expansion of write!
<std macros>:2:1: 2:46 note: expansion site
<std macros>:1:1: 4:66 note: in expansion of writeln!
src/main.rs:3:5: 3:34 note: expansion site
```

This can be pretty hairy to find if the project has many crates (my project euler repo, which is where I observed this, has 60+ executables).

**Comment by rprichard** [comment]
Related to #23459

**Comment by gbersac** [comment]
Still a problem with the 1.2 rustc compiler.

**Comment by pmarcelll** [comment]
Triage: we have nice error messages now.

**Comment by petrochenkov** [comment]
Current diagnostics:
```
rustc 1.17.0-nightly (b1e31766d 2017-03-03)
error: no method named `write_fmt` found for type `&mut std::vec::Vec<_>` in the current scope
 --> <anon>:3:5
  |
3 |     writeln!(&mut buffer, "foo").unwrap();
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: items from traits can only be used if the trait is in scope; the following trait is implemented but not in scope, perhaps add a `use` for it:
  = help: candidate #1: `use std::io::Write;`
  = note: this error originates in a macro outside of the current crate
```
